x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/CsvTests.java

```diff
@@ -149,6 +150,17 @@ public class CsvTests extends ESTestCase {
     private static final Logger LOGGER = LogManager.getLogger(CsvTests.class);
     private static final String IGNORED_CSV_FILE_NAMES_PATTERN = "-IT_tests_only";
 
+    private static final Set<NodeFeature> LIVE_FEATURES = new HashSet<>();
+    static {
+        EsqlFeatures features = new EsqlFeatures();
+        LIVE_FEATURES.addAll(features.getFeatures());
+        for (Map.Entry<NodeFeature, Version> e : features.getHistoricalFeatures().entrySet()) {
+            if (Version.CURRENT.onOrAfter(e.getValue())) {
```

Comment:
So, we want to eliminate references to `Version` outside of a few key infrastructure points. What is this aiming to do, and what do you expect `Version.CURRENT` to actually be?

Reply:
I'm trying to run all of the tests that have features declared on them that are supported by the code as it sits in the repo. In this place we're not running Elasticsearch proper. So I followed the same pattern that `FeatureService` uses. I spent a while trying to use `FeatureService` here but it would have required building a valid cluster state with the features in it. I don't know how to do that and figure that's at least as brittle as this. If `FeaturesService` allowed had a method like this I'd happily use it but I don't know if you want to grow that.

I don't really care what `Version.CURRENT` is here - just something that'll light up the historical features that have been defined to work on the current code.

Reply:
Maybe I can just drop this entirely and never test. I don't see any places where these csv tests turn off features because the version is too far ahead. So I imagine we don't want to disable any of these tests. Maybe I can just do that for now.

Reply:
That seems fine actually. I'll push that.

Reply:
`Version.CURRENT` will always be the current code version, so it will always use the maximal feature set declared. Do you actually want to check the behaviour when run against _specific_ previous versions?

Reply:
> If `FeaturesService` had a method like this I'd happily use it but I don't know if you want to grow that.

The obvious method is a `Set<NodeFeature> getApplicableHistoricalFeatures(Version)` method to get the features applicable at a specific version, but that requires specifying `Version`, which we don't want to do generally.

Reply:
Not here. I want to run against the maximal feature set. We run the csv-spec files in about 5 different ways. All of the other ways just get to use your standard version checks.

Reply:
OK - that makes sense